### PR TITLE
[Feature] enable Python API for auto tap changer (experimental)

### PR DESCRIPTION
**Pull request by mgovers**
Python equivalent of #597

### CODE REVIEW

File: src/power_grid_model/enum.py [hunk 1]
```diff
@@ -67,6 +67,27 @@ class CalculationMethod(IntEnum):
     iec60909 = 5
```
Comment (mgovers) [inline src/power_grid_model/enum.py:69]: TBD: should we add an enum `ControlSide` here?

```suggestion
class ControlSide(IntEnum):
    """Control Sides for Regulators on Branch Objects"""

    from_side = 0
    to_side = 1
    side_1 = 0
    side_2 = 1
    side_3 = 2


```

* [ ] If so: should we add a validation check to check that the from_side/to_side/side_1/side_2/side_3 is in range of the transformer type
* [ ] If not: should we add a validation check to check
  * [ ] that the correct enum type is being used and/or
  * [ ] that it is in range of the transformer type

**Comment by mgovers** [comment]
I believe this is all that's needed for the time being

File: src/power_grid_model/enum.py (hunk 1 above)
Comment (nitbharambe) [reply src/power_grid_model/enum.py:69]: I dont see an extra enum being useful. The intention is sufficiently conveyed by using BranchSide or Branch3Side itself.
A data validation check would be nice.

Comment (mgovers) [reply src/power_grid_model/enum.py:69]: Created a JIRA ticket for this

**Review by TonyXiang8787 — APPROVED** [review]

**Pull request merged**